src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -2080,8 +2080,7 @@ trait Types
     override def decls       = relativeInfo.decls
     override def bounds      = relativeInfo.bounds
 
-    // TODO: this deviates from the spec "The base types of an abstract type are the base types of its upper bound."
-    override protected[Types] def baseTypeSeqImpl: BaseTypeSeq = bounds.hi.baseTypeSeq prepend this
+    override protected[Types] def baseTypeSeqImpl: BaseTypeSeq = bounds.hi.baseTypeSeq
```

Comment:
@lrytz I snuck in another change -- I think the same reasoning applies here.

Reply:
Both Jenkins and Travis hate it

Reply:
I'll let this sleeping bug lie then. Poked around a bit in type inference, but nothing jumped out.